cmd/manager/operator.go

```diff
@@ -506,7 +510,20 @@ func ensureDefaultProfileBundles(
 			" Skipping defaults creation.")
 		return nil
 	}
+	arch := goruntime.GOARCH
+	archProducts, hasArch := disableProductPerArch[arch]
+	// build a map of disabled products for this arch
+	disabledProducts := make(map[string]bool)
```

Comment:
Instead of building a separate map, I wonder if we can reuse the `getValidPlatform` function by implementing the `goruntime.GOARCH` check there. Maybe something like:


```go
func getValidPlatform(p string) PlatformType {
        arch := goruntime.GOARCH
	switch {
	case strings.EqualFold(p, string(PlatformOpenShift)) && (arch == "ppc64le") :
		return PlatformOpenShiftOnPower
	case strings.EqualFold(p, string(PlatformOpenShift)) && (arch == "s390x") :
		return PlatformOpenShiftOnZ
	case strings.EqualFold(p, string(PlatformOpenShift)):
		return PlatformOpenShift
	case strings.EqualFold(p, string(PlatformEKS)):
		return PlatformEKS
	case strings.EqualFold(p, string(PlatformHyperShift)):
		return PlatformHyperShift
	case strings.EqualFold(p, string(PlatformGeneric)):
		return PlatformGeneric
	default:
		return PlatformUnknown
	}
}

```

Then we'd have somthing like:

```go
defaultProductsPerPlatform = map[PlatformType][]string{
	PlatformOpenShift: {
		"rhcos4",
		"ocp4",
	},
        PlatformOpenShiftOnPower: {
		"ocp4",
	},
        PlatformOpenShiftOnZ: {
		"ocp4",
	},
```

Then we continue to have a single mapping for default product support.

Reply:
I was thinking of having `disableProductPerArch` to work as a filter, will we have other platforms supported on P/Z in the future?

Reply:
What I also appreciate about the Lance's suggestion is the increased flexibility.
It also allows us to add/enable arch specific products as well.

Reply:
I think that makes sense, just updated it